eng/scripts/Validate-Typespec.ps1

```diff
@@ -0,0 +1,25 @@
+[CmdletBinding()]
+param (
+  [Parameter(Position = 0, Mandatory = $true)]
+  [string] $SpecsRepoRootDirectory,
+  [Parameter(Position = 1, Mandatory = $false)]
+  [string]$TargetBranch,
+  [Parameter(Position = 2, Mandatory = $false)]
+  [string]$SourceBranch
+)
+
+$typespecFolders = @()
+
+$typespecFolders = &"$PSScriptRoot/Get-Typespec-Folders.ps1" "$SpecsRepoRootDirectory" "$TargetBranch" "$SourceBranch"
+
+$exitCode = 0
+foreach ($typespecFolder in $typespecFolders) {
+  npx --no tsv $typespecFolder 2>&1 | Write-Host
+  if ($LASTEXITCODE) {
+    $exitCode = 1
+  }
+  git restore .
```

Comment:
Add another parameter to control whether the script runs `git restore` and `git clean` between folders.  In validation pipelines, these commands are good to isolate the diffs caused by each spec folder.  But on dev machines, you generally **want** to leave the files changed in your enlistment so you can view them in an editor, add them to a PR, etc.

Reply:
On a dev machine, if validating multiple specs without clean between specs, it means any diffs from earlier specs will cause later specs to fail validation.  But I think this is fine.  Users will likely review the git diffs in VS Code rather than from the command-line output.